### DISCUSSION TITLE
SOLVED: "Coapp not installed" VDH v8.2.1.1 (Debian) and Tor-browser

### DISCUSSION THREAD

**Discussion by GitHubrist** [aclap-dev/video-downloadhelper#239] [Help and Q&A]
### Your question

After being informed by the Video-DownloadHelper extension that it needed to have the "co-app" installed, I followed the instructions (for Linux, using the 'curl' command-line), but everytime afterward I tried to download a video, I was informed, again, that the co-app needed to be installed.

I checked[1] that, indeed, the co-app was not found by the extension.

After searching the web I found some clues, and proceeded to find:
  ~/.local/share/vdhcoapp/vdhcoapp[2]
  ~/.var/app/org.mozilla.firefox/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json[3]

It was not until I found another comment in these pages (I think) from a user who pointed out that 64-bit versions of Firefox use the directory /usr/lib64/mozilla/ but that the VDH extension continues to use 
/usr/lib/mozilla/ that I determined the solution.

SOLUTION
I created a directory: 
  /usr/lib64/mozilla/native-messaging-hosts/
and a link:
  sudo ln -s /usr/lib/mozilla /usr/lib64/mozilla
and copied the .json file into the new directory:
  sudo cp ~/.var/app/org.mozilla.firefox/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json /usr/lib64/mozilla/native-messaging-hosts/

After this step, the co-app was recognized and downloading was enabled.

[1] by going into Manage Extensions and clicking on the ellipsis(...) next to the VDH extension and choosing Preferences.
[2] the home of the co-app.
[3] apparently the substance of what tor-browser (firefox) needs to learn about the co-app.[4]
[4] for example, the contents of the .json (text) file contain the path to vdhcoapp.

### Addon & CoApp version and details

⊖
Video DownloadHelper 8.2.1.1 (stable)
Video DownloadHelper
Version 8.2.1.1 (stable)
Browser locale: en-US
Production build
Built on 2024-02-19
Build options: browser=firefox, linuxlic=false, noyt=false, target=mozilla
⊖
Platform
Platform Linux x86-64
Browser Tor Project Firefox 115.8.0
⊖
Companion App installed
Found companion app: VdhCoApp 2.0.19
Companion app binary: /home/alt/.local/share/vdhcoapp/vdhcoapp

**Comment by jcfrog** [comment]
Hi,
Did you read this? https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized#linux-all

**Reply by GitHubrist** [reply]
Yes. I did.  
However 'vdhcoapp install' did not perform the steps I did.

**Reply by jcfrog** [reply]
@paulrouget can you take a look?

**Comment by paulrouget** [comment]
I'm surprised you had to do all of that.

Normally, a file has been created here:

`~/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json`

and that should have been enough.

**Comment by GitHubrist** [comment]
The file you named was present, but the add-on did not recognize the co-app
as installed--even after exiting and restarting Firefox.

On Mon, Mar 4, 2024 at 12:00 AM Paul ***@***.***> wrote:

> I'm surprised you had to do all of that.
>
> Normally, a file has been created here:
>
> ~/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
>
> and that should have been enough.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aclap-dev/video-downloadhelper/discussions/239#discussioncomment-8662400>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEM4BZKO5VHZCDBVOI6X53DYWQEWDAVCNFSM6AAAAABEB2UMJWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMNRSGQYDA>
> .
> You are receiving this because you authored the thread.Message ID:
> <aclap-dev/video-downloadhelper/repo-discussions/239/comments/8662400@
> github.com>
>


-- 
*Cheer Up!*

**Comment by anibalisola** [comment]
I've the same problem, but i couldn't resolve.
On Firefox (125.0.3) it works well, but on tor (Firefox 115.10.oesr) no, running on Fedora 39 (before) and now 40.
Since the lust VDH update, something has occurred.
I did the instructions of @GitHubrist but in Fedora it didn't work.

**Comment by bobblkabb** [comment]
Same here but on openSUSE Tumbleweed.

**Comment by lubboo** [comment]
Here is the solution for my **Tor-Browser** (Linux Mint)

Copy the file “net.downloadhelper.coapp.json” from /home/user/.mozilla/native-messaging-hosts/ to /home/user/tor-browser_en-US/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts/.
The download helper application will then be found without restarting.

Source: https://github.com/torproject/torbrowser-launcher/issues/301#issuecomment-422809589

**Comment by robvanson** [comment]
I had the same problem with Tor on Mac (Sonoma, 14.5). VDH worked fine under Firefox and every other browser, but not on the Tor browser (M2 Macbook).

I solved it by copying the json file: 
> ~/Library/Application\ Support/Mozilla/NativeMessagingHosts/net.downloadhelper.coapp.json
to: 
> ~/Library/Application\ Support/TorBrowser-Data/Browser/mozilla/NativeMessagingHosts/

Thus:
> mkdir -p ~/Library/Application\ Support/TorBrowser-Data/Browser/mozilla/NativeMessagingHosts
> cp ~/Library/Application\ Support/Mozilla/NativeMessagingHosts/net.downloadhelper.coapp.json \\
> ~/Library/Application\ Support/TorBrowser-Data/Browser/mozilla/NativeMessagingHosts/

And it finally worked.

**Comment by lfg-max** [comment]
Read this.

https://github.com/aclap-dev/video-downloadhelper/discussions/912


This is a better solution and a simple file copy from. one location to another.
